go-testing-tutorial/main_test.go

```diff
@@ -138,3 +139,24 @@ func TestHttp(t *testing.T) {
 		t.Error("Test failed. Expected status to equal expected service response.")
 	}
 }
+
+func TestSearchHandlerShouldReturn404IfNoSearchQueryIsPresent(t *testing.T) {
+	req, err := http.NewRequest("GET", "/", nil)
+	if err != nil {
+		return
+	}
+	w := httptest.NewRecorder()
+	searchHandler(w, req)
+
+	resp := w.Result()
+	body, _ := ioutil.ReadAll(resp.Body)
+
+	if resp.StatusCode != 400 {
```

Comment:
This is the wrong status code to check for in this case. I believe you want to check for 200.

Reply:
I should add a string query to make the status code 200?